frappe/www/printview.py

```diff
@@ -52,21 +52,39 @@ def get_context(context):
 		doctype=frappe.form_dict.doctype, document=frappe.form_dict.name, file_type="PDF", method="Print"
 	)
 
-	return {
-		"body": get_rendered_template(
+	is_invalid_print = False
+	print_style = None
+	try:
+		body = get_rendered_template(
 			doc,
 			print_format=print_format,
 			meta=meta,
 			trigger_print=frappe.form_dict.trigger_print,
 			no_letterhead=frappe.form_dict.no_letterhead,
 			letterhead=letterhead,
 			settings=settings,
-		),
-		"css": get_print_style(frappe.form_dict.style, print_format),
+		)
+		print_style = get_print_style(frappe.form_dict.style, print_format)
+	except frappe.exceptions.LinkExpiredError:
+		body = frappe.get_template("templates/print_formats/print_key_expired.html").render({})
+		context.http_status_code = 410
+		is_invalid_print = True
+	except frappe.exceptions.InvalidKey:
+		body = frappe.get_template("templates/print_formats/print_key_invalid.html").render({})
+		context.http_status_code = 401
+		is_invalid_print = True
+
+	return {
+		"body": body,
+		"print_style": print_style,
```

Comment:
let's not change the key `css`, can cause compatibility issue (?)

Reply:
> can cause compatibility issue (?)

How 🤔 

"css" was explicitly built in "printview" and passed... I have refactored the usage as well... I don't think there was a way to extend or override that part.

Reply:
You're right. `get_context` is only used internally.